### ISSUE TITLE
4.0.0: Payment proof is not accessible on receiver side

### ISSUE THREAD

**Issue by lehnberg** [mimblewimble/grin-wallet#441]
## Steps to reproduce
1. Grin-wallet 4.0.0-beta2 both sides.
2. Send slatepack tx with payment proof (`-y`).
3. `grin-wallet txs` on both sender and receiver side.

## Expected behaviour
As the payment proof proves that both parties of a transaction have participated (sender sent to receiver), it should be accessible in the wallets of both sender and the receiver.

## Actual behaviour
The sender's `Payment Proof` cell reads `Yes`, the receiver's `Payment Proof` cell reads `none`.

**Comment by yeastplume** [comment]
This has always been the case, by design. The sender is the one who completes the proof, and can optionally provide it to the recipient after the fact:

https://github.com/mimblewimble/grin-rfcs/blob/master/text/0006-payment-proofs.md

**Comment by lehnberg** [comment]
I see, thanks for that. Isn't that column a bit misleading in that case? 

The recipient's wallet would be aware that a payment proof was in fact requested (as it provided `receiver_signature` and sent to the sender). Perhaps the data on the recipient side would need to reflect that somehow? `None` suggests that there was no payment proof at all.

**Comment by johndavies24** [comment]
I agree with @lehnberg, this was confusing to me and if over tor it should be simple to send the payment proof to recipient.  At the very least, maybe indicate that a payment proof was apart of the tx

**Comment by ardocrat** [comment]
Payment proof can be generated by sender after tx will be confirmed at blockchain, currently it can be done only manually, we can add API to request payment proof by receiver (over Tor for now) it requires sender to be online (what can create additional complexity) but we can give it a try.
